rasa/nlu/test.py

```diff
@@ -664,9 +633,53 @@ def evaluate_intents(
         "precision": precision,
         "f1_score": f1,
         "accuracy": accuracy,
+        "errors": intent_errors,
     }
 
 
+def _calculate_report(
+    filename: Text,
+    output_directory: Optional[Text],
+    targets: Iterable[Any],
+    predictions: Iterable[Any],
+    report_as_dict: bool = False,
+    exclude_label: Optional[Text] = None,
+) -> Tuple[Union[Text, Dict], float, float, float, np.ndarray, List[Text]]:
+    from rasa.test import get_evaluation_metrics
+    import sklearn.metrics
+    import sklearn.utils.multiclass
+
+    confusion_matrix = sklearn.metrics.confusion_matrix(targets, predictions)
+    labels = sklearn.utils.multiclass.unique_labels(targets, predictions)
+
+    if output_directory:
+        report, precision, f1, accuracy = get_evaluation_metrics(
+            targets, predictions, output_dict=True, exclude_label=exclude_label
+        )
+        report = _add_confused_labels_to_report(
+            report,
+            confusion_matrix,
+            labels,
+            exclude_labels=[exclude_label] if exclude_label else [],
+        )
+
+        report_filename = os.path.join(output_directory, filename)
+        rasa.shared.utils.io.dump_obj_as_json_to_file(report_filename, report)
```

Comment:
feels to me like file dumping shouldn't be part of the method `_calculate_report`.